documentation/src/main/asciidoc/ch11.asciidoc

```diff
@@ -694,17 +694,25 @@ mechanism to register additional constraint definitions. All you have to do is t
 _javax.validation.ConstraintValidator_ to _META-INF/services_. In this service file you list the
 fully qualified classnames of your constraint validator classes (one per line). Hibernate Validator
 will automatically infer the constraint types they apply to.
-See <<example-using-service-file-for-constraint-definitions>> for an example.
+See <<example-using-service-file-for-constraint-definitions,Constraint definition via service file>>
+for an example.
 
 [[example-using-service-file-for-constraint-definitions]]
-.Using _META-INF/services/javax.validation.ConstraintValidator_ for constraint definitions
+._META-INF/services/javax.validation.ConstraintValidator_
+
 ====
 [source]
 ----
+# Assuming a custom constraint annotation @org.mycompany.CheckCase
 org.mycompany.CheckCaseValidator
 ----
 ====
 
+To contribute default messages for your custom constraints, place a _ValidationMessages.properties_
```

Comment:
> Hmm, are we not running into a visibility problem again? What if I add more than one contributor jar each providing a ContributorValidationMessages?

That's what I meant to address when saying "which then is 'merged' (e.g. similar to service loader files)". I.e. we'd consider all the entries from all the `ContributorValidationMessages` bundles. 

There might be a collission of keys, but best practise is to give them "fully-qualified" so it shouldn't be a problem in practice. I guess the same issue btw. exists with the original proposal when giving a bundle in the same package in two different archives.